arch/x86/core/ia32/cache.c

```diff
@@ -47,15 +47,15 @@ extern size_t z_cache_line_size_get(void);
  * @return N/A
  */
 
-_sys_cache_flush_sig(_cache_flush_clflush)
+void arch_dcache_flush(void *start_addr, size_t size)
 {
-	int end;
+	uintptr_t end;
```

Comment:
Suggestion: shall we do the same thing we did for ARC here just to make it a bit prettier?
```
void arch_dcache_flush(void *start_addr, size_t size)
{
	uintptr_t start = (uintptr_t)start_addr;
	...
```